### ISSUE TITLE
Bug: NumberFormatter::setAttribute Passing null to parameter #2 ($value) of type int|float is deprecated

### ISSUE THREAD

**Issue by bedrijfsportaal** [codeigniter4/CodeIgniter4#7156]
### PHP Version

8.0

### CodeIgniter4 Version

4.3.1

### CodeIgniter4 Installation Method

Composer (using `codeigniter4/appstarter`)

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

cli-server (PHP built-in webserver)

### Database

MariaDB 10.4.24

### What happened?

Using the service::render with `{ price|local_currency(eur,nl_NL) }` is not working, the fraction option cannot be set and doesn't accept `null` value.

**Error** NumberFormatter::setAttribute(): Passing null to parameter #2 ($value) of type int|float is deprecated

### Steps to Reproduce

Render view with `{ price|local_currency(eur,nl_NL) }` as option.

### Expected Output

Be able to set the fraction like `{ price|local_currency(eur,2,nl_NL) }`

**Comment by bedrijfsportaal** [comment]
Found the issue for my problem, the documentation is missing a third parameter, it's allowed to use the following way `{ price|local_currency(eur,nl_NL,2) }` so the fraction is set. But the issue remains if you don't set the third parameter, the error will still be thrown.

**Comment by kenjis** [comment]
Thank you for reporting.
I sent a PR: #7157

**Comment by kenjis** [comment]
> Missing the option to set the default view path on initialize as well, had to find out by digging to the code that it's allowed to change the view path.

What do you mean?
Why do you need to change the view path?

**Comment by erikkraijenoord** [comment]
> > Missing the option to set the default view path on initialize as well, had to find out by digging to the code that it's allowed to change the view path.
> 
> What do you mean?
> Why do you need to change the view path?
> 

Because of the folder structure I use, i'm working in modules so below is my setup.

App
-- Modules
--- ModuleName
---- Views

Bij default it's set to read from

App
-- Views

**Comment by bedrijfsportaal** [comment]
In my case i work from a modules folder inside the app folder, example:

App
-- Modules
---- Modulename
------- Views

By default it looks at: 

App
-- Views

**Comment by kenjis** [comment]
You can use Namespaced Views.
See https://codeigniter4.github.io/CodeIgniter4/outgoing/views.html#namespaced-views

But you mean you want to write code like `view('blog_view')`?

**Comment by bedrijfsportaal** [comment]
> You can use Namespaced Views. See https://codeigniter4.github.io/CodeIgniter4/outgoing/views.html#namespaced-views
> 
> But you mean you want to write code like `view('blog_view')`?

I think it's a misunderstanding from my side, i'm using the https://codeigniter4.github.io/CodeIgniter4/outgoing/view_renderer.html#CodeIgniter\View\View that doesn't work with Namespaces.

**Comment by kenjis** [comment]
Did you define the namespace for the module?
See https://codeigniter4.github.io/CodeIgniter4/general/modules.html#namespaces

`view()` calls ` CodeIgniter\View\View::render()` inside. 
So Namespaced Views also should work.

**Comment by kenjis** [comment]
I mean namespaced view is like `view('Example\Blog\Views\blog_view')`.
So in your case, `return $render->setData(['key' => 'val'])->render('App\Modules\ModuleName\Views\view_name')`

**Comment by bedrijfsportaal** [comment]
> I mean namespaced view is like `view('Example\Blog\Views\blog_view')`. So in your case, `return $render->setData(['key' => 'val'])->render('App\Modules\ModuleName\Views\view_name')`

This in my case wil result in a ErrorException and by default looks at `App\Views\view_name.php` when using:

```php
// Will by default look in: App\Views\view_name.php
return $render->setData(['key' => 'val'])->render('view_name');

**Comment by kenjis** [comment]
No, I mean:
```php
return $render->setData(['key' => 'val'])->render('App\Modules\ModuleName\Views\view_name');
```

**Comment by bedrijfsportaal** [comment]
> No, I mean:
> 
> ```
> return $render->setData(['key' => 'val'])->render('App\Modules\ModuleName\Views\view_name');
> ```

That will result in the error given below:

```text
CodeIgniter\View\Exceptions\ViewException
Invalid file: "C:\***\app\Config/../Views\App\Modules\Bookings\Views\order_template.php"

**Comment by kenjis** [comment]
@bedrijfsportaal Cannot reproduce.

The following code works fine:
https://github.com/kenjis/ci4-modules-test/blob/44ecb00875aead3d240f09f81f50338ce94dec3e/modules/Home/src/Controllers/Home.php#L37

This repository code does not work on your Windows?
https://github.com/kenjis/ci4-modules-test
Novigate to http://localhost:8080/parser

**Comment by bedrijfsportaal** [comment]
> @bedrijfsportaal Cannot reproduce.
> 
> The following code works fine: https://github.com/kenjis/ci4-modules-test/blob/44ecb00875aead3d240f09f81f50338ce94dec3e/modules/Home/src/Controllers/Home.php#L37
> 
> This repository code does not work on your Windows? https://github.com/kenjis/ci4-modules-test Navigate to http://localhost:8080/parser

My folder structure differs from your example, the only way this is working is with:
```php
$render = Services::parsert( '\App\Modules\ModuleName' );
```

Without it will look inside my default Views folder, i don't believe it's a bug because i've got an option to override this and fix my issue.

**Comment by kenjis** [comment]
Good to hear.

As you say, the following is not documented.

```php
$render = Services::parser('\App\Modules\ModuleName');
$render = Services::renderer('\App\Modules\ModuleName');
```

It seems better to be documented in the user guide.

**Comment by kenjis** [comment]
Ah, it seems the following code worked, but it is still a misuse.
Because the `$viewPath` is the directory, not a namespace.
```php
$render = Services::parser( '\App\Modules\ModuleName' );
```
https://github.com/codeigniter4/CodeIgniter4/blob/29c31173b6ecdd3be75ea69b3f5eb876e7b5dcbe/system/View/View.php#L44-L48

**Comment by kenjis** [comment]
Added sample code:
https://github.com/kenjis/ci4-modules-test/blob/88c633af55cf2e0460db8cca9fa8e4f6f6a031e2/modules/Home/src/Controllers/Home.php#L42

**Comment by bedrijfsportaal** [comment]
> Ah, it seems the following code worked, but it is still a misuse.
> 
> ```
> $render = Services::parser( '\App\Modules\ModuleName' );
> ```
> 
> Because the `$viewPath` is the directory, not a namespace.
> 
> https://github.com/codeigniter4/CodeIgniter4/blob/29c31173b6ecdd3be75ea69b3f5eb876e7b5dcbe/system/View/View.php#L44-L48

Ah dammit, your right totally overlooked that part...
Using it as a file path does the trick as well like you've mentioned.

Thank you for the clearifycation, will keep this in mind for other modules!

**Comment by kenjis** [comment]
I found a bug that the following does not work, because the debugbar initialize the shared renderer service.

```php
$render = Services::renderer('/path/to/Views');
```

I sent a PR to fix #7172

**Comment by bedrijfsportaal** [comment]
> I found a bug that the following does not work, because the debugbar initialize the shared renderer service.
> 
> ```
> $render = Services::renderer('/path/to/Views');
> ```
> 
> I sent a PR to fix #7172

Debugbar is disabled by default for me, so i wouldn't have noticed that, but good to hear the bug is as quickly fixed as it is found!
